torchgeo/transforms/indices.py

```diff
@@ -0,0 +1,234 @@
+
+# https://github.com/pytorch/pytorch/issues/60979
+# https://github.com/pytorch/pytorch/pull/61045
+Module.__module__ = "torch.nn"
+
+
+epsilon = 1e-10
+
+
+def ndbi(swir: Tensor, nir: Tensor) -> Tensor:
+    """Compute Normalized Different Built-up Index (NDBI).
+
+    Args:
+        swir: tensor containing swir band
+        nir: tensor containing nir band
+
+    Returns:
+        tensor containing computed NDBI values
+    """
+    return (swir - nir) / ((swir + nir) + epsilon)
+
+
+def ndsi(green: Tensor, swir: Tensor) -> Tensor:
+    """Compute Normalized Different Snow Index (NDSI).
+
+    Args:
+        green: tensor containing green band
+        swir: tensor containing swir band
+
+    Returns:
+        tensor containing computed NDSI values
+    """
+    return (green - swir) / ((green + swir) + epsilon)
+
+
+def ndvi(red: Tensor, nir: Tensor) -> Tensor:
+    """Compute Normalized Different Vegetation Index (NDVI).
+
+    Args:
+        red: tensor containing red band
+        nir: tensor containing nir band
+
+    Returns:
+        tensor containing computed NDVI values
+    """
+    return (nir - red) / ((nir + red) + epsilon)
+
+
+def ndwi(green: Tensor, nir: Tensor) -> Tensor:
+    """Compute Normalized Different Water Index (NDWI).
+
+    Args:
+        green: tensor containing green band
+        nir: tensor containing nir band
+
+    Returns:
+        tensor containing computed NDWI values
+    """
+    return (green - nir) / ((green + nir) + epsilon)
+
+
+class AppendNDBI(Module):  # type: ignore[misc,name-defined]
+    """Normalized Difference Built-up Index (NDBI).
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://doi.org/10.1080/01431160304987
+    """
+
+    def __init__(self, index_swir: int, index_nir: int, dim: int = 1) -> None:
+        """Initialize a new transform instance.
+
+        Args:
+            index_swir: index of the Short-wave Infrared (SWIR) band in the image
+            index_nir: index of the Near Infrared (NIR) band in the image
+
+            dim: dimension of channels in the input tensors (default: 1)
+        """
+        super().__init__()
+        self.dim = dim
+        self.index_nir = index_nir
+        self.index_swir = index_swir
+
+    def forward(self, sample: Dict[str, Tensor]) -> Dict[str, Tensor]:
+        """Create a band for NDBI and append to image channels.
+
+        Args:
+            sample: a single data sample
+
+        Returns:
+            a sample where the image has an additional channel representing NDBI
+        """
+        if "image" in sample:
+            index = ndbi(
+                swir=sample["image"][:, self.index_swir],
+                nir=sample["image"][:, self.index_nir],
+            )
+            index = index.unsqueeze(self.dim)
+            sample["image"] = torch.cat([sample["image"], index], dim=self.dim)  # type: ignore[attr-defined]  # noqa: E501
```

Comment:
If you split this into multiple lines you won't need to ignore line length here. Like:
```python
sample["image"] = torch.cat(  # type: ignore[attr-defined]
    ...
)
```